pkg/gds/admin.go

```diff
@@ -1006,6 +1007,133 @@ func (s *Admin) Resend(c *gin.Context) {
 	c.JSON(http.StatusOK, out)
 }
 
+// ReviewTimeline returns a list of time series records containing registration state counts by week.
+func (s *Admin) ReviewTimeline(c *gin.Context) {
+	// Go needs this constant to determine the time format
+	const timeFormat = "2006-01-02"
+	var (
+		err          error
+		in           *admin.ReviewTimelineParams
+		out          *admin.ReviewTimelineReply
+		startTime    time.Time
+		endTime      time.Time
+		earliestTime time.Time
+		weekTime     time.Time
+		numWeeks     int
+		vaspCounts   []map[string]bool
+	)
+
+	// Get request parameters
+	in = new(admin.ReviewTimelineParams)
+	if err = c.ShouldBindQuery(&in); err != nil {
+		log.Warn().Err(err).Msg("could not bind request with query params")
+		c.JSON(http.StatusBadRequest, admin.ErrorResponse(err))
+		return
+	}
+
+	// Default value for start date
+	earliestTime = time.Date(2020, time.January, 1, 0, 0, 0, 0, time.Local)
+	if in.Start != "" {
+		// Parse start date
+		if startTime, err = time.Parse(timeFormat, in.Start); err != nil {
+			log.Warn().Err(err).Msg("could not parse start date")
+			c.JSON(http.StatusBadRequest, admin.ErrorResponse(fmt.Errorf("invalid start date: %s", in.Start)))
+		}
+		// Hard limit on the earliest date to avoid making the server do unnecessary work
+		if startTime.Before(earliestTime) {
+			startTime = earliestTime
+		}
```

Comment:
I totally understand what you're getting at with this hard limit, but since you're defaulting to `earliestTime` and the user is submitting the request for this time, it's probably best not restrict them artificially -- even if the user submits epoch time you're still talking about only ~3500 weeks, which isn't terrible as a for loop. Perhaps you could `log.Warn()` if the number of weeks is > 260 (e.g. five years)?

Reply:
My thinking was that the earliest date a user can technically request is 0000-01-01 which is ~100,000 weeks. Maybe a solution could be to change the hard limit to the epoch time or return an error if the given date is earlier than the epoch?

Reply:
Ok, I'm on board with ensuring that it's later than epoch time - that's an understandable time limit that users will understand naturally.